javalib/src/main/scala/java/lang/String.scala

```diff
@@ -831,8 +831,61 @@ final class _String()
     preprocessed.toLowerCase()
   }
 
-  @inline
-  def toLowerCase(): _String = toCase(Character.toLowerCase)
+  @inline def toLowerCase(): _String = {
+    replaceCharsAtIndex { i =>
+      /* Tests whether we are in an `Final_Sigma` context.
+       * From Table 3.17 in the Unicode standard:
+       * - Description: C is preceded by a sequence consisting of a cased letter and then zero or more case-ignorable characters,
+       *     and C is not followed by a sequence consisting of zero or more case-ignorable characters and then a cased letter.
+       * - Regex:
+       *     before C: \p{cased}(\p{case-ignorable})*
+       *     after C:  !((\p{case-ignorable})*\p{cased})
+       */
+      def isFinalSigma(idx: Int): scala.Boolean = {
+        import Character._
+
+        val hasCasedBefore = {
+          val j = skipCaseIgnorableCharsBackwards(idx)
+          j > 0 && isCased(this.codePointBefore(j)) ||
+          j == 0 && isCased(this.codePointAt(j))
```

Comment:
I think the last condition will always return false. If `this.codePointAt(0)` were `isCased`, it cannot be `isCaseIgnorable` at the same time, and therefore `skipCaseIgnorableCharsBackwards` would not have skipped it. So it would have returned a `j > 0`.
```suggestion
          val j = skipCaseIgnorableCharsBackwards(idx)
          j > 0 && isCased(this.codePointBefore(j))
```